addons/binding/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/discovery/Enigma2DiscoveryParticipant.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.enigma2.internal.discovery;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant;
+import org.openhab.binding.enigma2.Enigma2BindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Enigma2DiscoveryParticipant} is responsible processing the
+ * results of searches for mDNS services of type _ssh._tcp.local. and finding a webinterface
+ *
+ * @author Thomas Traunbauer - Initial contribution
+ */
+public class Enigma2DiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private Logger logger = LoggerFactory.getLogger(Enigma2DiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Enigma2BindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo info) {
+        DiscoveryResult result = null;
+        ThingUID uid = getThingUID(info);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>(4);
+            String label;
+            try {
+                label = info.getName();
+            } catch (Exception e) {
+                label = "unnamed enigma2 device";
```

Comment:
Can/will this happen, do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception